src/lib.rs

```diff
@@ -229,6 +302,68 @@ impl IRCompiler {
         }
     }
 
+    #[allow(clippy::too_many_arguments)]
+    pub fn set_ray_tracing_pipeline_arguments(
+        &mut self,
+        max_attribute_size_in_bytes: u32,
+        raytracing_pipeline_flags: ffi::IRRaytracingPipelineFlags,
+        chs: u64,
+        miss: u64,
+        any_hit: u64,
+        callable_args: u64,
+        max_recursive_depth: i32,
+    ) {
+        unsafe {
+            self.funcs.IRCompilerSetRayTracingPipelineArguments(
+                self.me,
+                max_attribute_size_in_bytes,
+                raytracing_pipeline_flags,
+                chs,
+                miss,
+                any_hit,
+                callable_args,
+                max_recursive_depth,
+            )
+        }
+    }
+
+    pub fn set_hitgroup_type(&mut self, hit_group_type: ffi::IRHitGroupType) {
+        unsafe {
+            self.funcs
+                .IRCompilerSetHitgroupType(self.me, hit_group_type)
+        }
+    }
+
+    pub fn synthesize_indirect_intersection_function(
+        &mut self,
+    ) -> Result<IRMetalLibBinary, CompilerError> {
+        let target_metallib = IRMetalLibBinary::new(self);
+        if unsafe {
+            self.funcs
+                .IRMetalLibSynthesizeIndirectIntersectionFunction(self.me, target_metallib.me)
+        } {
+            return Ok(target_metallib);
+        }
+        {
+            Err(CompilerError::SynthesizeIndirectIntersectionFunctionError)
+        }
+    }
+
+    pub fn synthesize_indirect_ray_dispatch_function(
+        &mut self,
+    ) -> Result<IRMetalLibBinary, CompilerError> {
+        let target_metallib = IRMetalLibBinary::new(self);
+        if unsafe {
+            self.funcs
+                .IRMetalLibSynthesizeIndirectRayDispatchFunction(self.me, target_metallib.me)
+        } {
+            return Ok(target_metallib);
+        }
+        {
+            Err(CompilerError::SynthesizeIndirectRayDispatchError)
```

Comment:
Not sure if `None` is more clear? As discussed before these functions are a bit weird because they return a `enum CompilerError` but can really only return one of the variants... Hence they shouldn't return that `enum`.

Reply:
Replaced it with Option, so CompilerError is now a struct instead of an enum